epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/MG5aMC_patches/PROD/patch.common

```diff
@@ -336,14 +381,11 @@ index 8c509e83f..cb6bf4ca5 100755
 +
 +                    cudacpp_backend = self.run_card['cudacpp_backend'] # the default value is defined in banner.py
 +                    logger.info("Building madevent in madevent_interface.py with '%s' matrix elements"%cudacpp_backend)
-+                    if cudacpp_backend == 'FORTRAN':
-+                        self.compile(['madevent_fortran_link'], cwd=Pdir)
-+                    elif cudacpp_backend == 'CPP':
-+                        self.compile(['madevent_cpp_link'], cwd=Pdir)
-+                    elif cudacpp_backend == 'CUDA':
-+                        self.compile(['madevent_cuda_link'], cwd=Pdir)
++                    cudacpp_supported_backends = [ 'fortran', 'cuda', 'hip', 'cpp', 'cppnone', 'cppsse4', 'cppavx2', 'cpp512y', 'cpp512z', 'cppauto' ]
++                    if cudacpp_backend in cudacpp_supported_backends :
++                        self.compile(['madevent_' + cudacpp_backend + '_link'], cwd=Pdir)
 +                    else:
-+                        raise Exception("Invalid cudacpp_backend='%s': only 'FORTRAN', 'CPP', 'CUDA' are supported")
++                        raise Exception( "Invalid cudacpp_backend='%s': supported backends are %s"%supported_backends )
 +                        ###logger.info("Building madevent with ALL (FORTRAN/CPP/CUDA) matrix elements (cudacpp_backend=%s)"%cudacpp_backend)
```

Comment:
Those change sounds quite dangerous to be put in this file since user might bypass them...
So I/We need to move those change in a more stable place (i.e. in a file from the PLUGIN)

Reply:
Hi @oliviermattelaer thanks. I have added many comments about this in a new issue #844. My suggestion is to handle this as a separate issue - without blocking the merge of this PR, which does not introduce or amplify the problem in my opinion.